homeassistant/components/media_player/enigma2.py

```diff
@@ -0,0 +1,220 @@
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    MediaPlayerDevice, PLATFORM_SCHEMA, MEDIA_TYPE_TVSHOW)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON, STATE_UNKNOWN, CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['openwebif.py==0.9']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TIMEOUT = 'timeout'
+
+DEFAULT_NAME = 'Enigma2 Media Player'
+DEFAULT_PORT = 80
+DEFAULT_TIMEOUT = 0
+
+# Support different items when watching recording playback
+SUPPORT_ENIGMA2_RECORDING_PLAYBACK = SUPPORT_VOLUME_SET | \
+                                     SUPPORT_VOLUME_MUTE | \
+                                     SUPPORT_TURN_OFF | SUPPORT_PAUSE
+
+SUPPORT_ENIGMA2_LIVE_TV = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+                          SUPPORT_TURN_OFF | SUPPORT_NEXT_TRACK | \
+                          SUPPORT_PREVIOUS_TRACK | SUPPORT_TURN_ON
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Enigma2 TV platform."""
+    name = config.get(CONF_NAME)
+
+    # Generate a configuration for the Enigma2 library
+    remote_config = {
+        'name': 'HomeAssistant',
+        'description': config.get(CONF_NAME),
+        'port': config.get(CONF_PORT),
+        'host': config.get(CONF_HOST),
+        'timeout': config.get(CONF_TIMEOUT),
+    }
+
+    add_devices([Enigma2Device(name, remote_config)])
+
+
+# pylint: disable=abstract-method
+class Enigma2Device(MediaPlayerDevice):
+    """Representation of a Enigma2 box."""
+
+    # pylint: disable=too-many-public-methods
+    def __init__(self, name, config):
+        """Initialize the Enigma2 device."""
+        import openwebif.api
+        self._state = STATE_UNKNOWN
+        # self.config = config
+
+        self._name = name
+        self.e2_box = openwebif.api.CreateDevice(config['host'],
+                                                 port=config['port'])
+
+        self.volume = 20
+        self.current_service_channel_name = None
+        self.current_programme_name = None
+        self.currservice_serviceref = None
+        self.muted = False
+        self.picon_url = None
+
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        from openwebif.api import PlaybackType
+
+        if self.e2_box.get_current_playback_type(
```

Comment:
Supported features should not change during the lifetime of an entity.

See https://github.com/home-assistant/architecture/issues/1.